src/frontend/pages/videoListing/VideoListing.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import axios from "axios";
+import "./videolisting.css";
+import VideoScroll from "./VideoScroll";
+import { useActionContext, useAuthContext } from "../../context";
+import {
+	deleteWatchLaterData,
+	postLikeData,
+	deleteLikesData,
+	postWatchLaterData,
+} from "../../services";
+
+function VideoListing() {
+	const { videoId } = useParams();
+	const [video, setVideo] = useState("");
+
+	const { actionState, actionDispatch } = useActionContext();
+	const { dataLikes, dataWatchLater } = actionState;
+	const {
+		authState: { token },
+	} = useAuthContext();
+
+	useEffect(() => {
+		(async () => {
```

Comment:
you can make a separate function for api (just for suggestion)

Reply:
yes i will add it